hphp/runtime/ext/pdo/ext_pdo.cpp

```diff
@@ -2037,7 +2038,31 @@ static bool do_fetch(sp_PDOStatement stmt,
 
     case PDO_FETCH_CLASS:
       if ((flags & PDO_FETCH_SERIALIZE) == 0 || idx) {
-        ret.toObject()->o_set(name, val);
+
+        if (ret.getObjectData()->instanceof(s_PDOEntity)) {
+          if (!stmt->support(PDOStatement::MethodGetColumnMeta)) {
+            pdo_raise_impl_error(stmt->dbh, stmt, "IM001",
+                                 "driver doesn't support meta data");
+            return false;
+          }
+
+          setPDOErrorNone(stmt->error_code);
+          Array meta;
+
+          if (!stmt->getColumnMeta(idx, meta)) {
+            PDO_HANDLE_STMT_ERR(stmt);
+            return false;
+          }
+            
+          Array params = Array::Create();
+          params.append(name);
+          params.append(val);
+          params.append(meta);
+          ret.asCObjRef()->o_invoke("setfield", params);
```

Comment:
o_invoke looks simple because it's lacking important inputs, hence its deprecation. Sorry, internal task number, but it says exactly what's in the comment, there's no extra context there.

The defaults for those params (null, null, and normal) should be fine for this use case. The key difference is that you'll need to setup a CallCtx to make the call.